tests/_util/test_functions.py

```diff
@@ -67,3 +69,25 @@ def test_validate_type_set_type():
 def test_object_to_link_by_uid_missing_uids():
     assert {'foo': 'bar'} == object_to_link_by_uid({'foo': 'bar'})
 
+
+def test_rewrite_s3_links_locally():
+    assert "http://localhost:9572" == rewrite_s3_links_locally("http://localstack:4572")
+
+@patch("os.path.isdir")
+@patch("os.path.join")
+@patch("os.makedirs")
+def test_write_file_locally(mock_makedirs, mock_join, mock_isdir):
+    mock_join.return_value = "/User/is/fake/myfile.pdf"
+    with patch('os.path.split') as mock_split:
```

Comment:
Why is it necessary to mock the `os.path.split` method? It's getting `"/User/is/fake/myfile.pdf"`, which it should be able to split.

Reply:
Yeah that worked fine: good catch!